wp-content/plugins/ig-wp-api-extensions/endpoints/v3/APIv3_Posts_Abstract.php

```diff
@@ -180,44 +208,47 @@ protected function get_changed_posts(WP_REST_Request $request, Array $all_posts)
 	}
 
 	/*
-	 * Validate the $posts by checking whether its elements meet the following format:
-	 *     Array => (
-	 *         'id' => Integer,
-	 *         'hash' => String
-	 *     )
+	 * Sort, cast and filter the keys of every $post contained in $posts
 	 */
-	private function validate_posts($posts) {
+	private function sanitize_posts($posts) {
+		if (!is_array($posts)) { // If no array was given, we can't do anything
+			return false;
+		}
+		if (isset($posts[0]['id'])) { // Check if the first element has either 'id' or 'path' key
```

Comment:
I guess it would be safer, to check this for each post. But if we include this caveat in the definition of the API, I'm fine with it ;-)

Reply:
In fact, I check it for every post [in this line](https://github.com/Integreat/cms/pull/894/files#diff-b747a3da16cb7c152e0ba2ecd8422ab8R227). I just use the first post to determine the key type used for the check of all posts. So if the first post has the key 'id', all other post have to have this key as well and the same holds for 'path'. Even though it makes the check a bit cumbersome, the diff calculation is cleaner if 'id' and 'path' are not mixed in one request.